improver/spotdata/spot_manipulation.py

```diff
@@ -148,12 +158,35 @@ def process(self, cubes: CubeList) -> Cube:
         neighbour_cube = cubes[-1]
         cube = cubes[0]
 
-        if self.realization_collapse:
-            cube = collapse_realizations(cube)
+        # If a spot forecast cube is passed as input constrain the sites to
+        # those that are found in the neighbour cube if an ID coordinate on
+        # which to constrain is provided, e.g. wmo_id. Otherwise pass the
+        # spot forecast cube forwards unchanged.
+        if cube.coords("spot_index"):
```

Comment:
I think that the main ambiguity currently with this implementation is with regards to the interaction between the subsetting of an existing site cube with the apply lapse rate functionality. I think that the expectation is that lapse rate correction is not applied to pre-existing site cubes, although I think that this option isn't explicity handled at the moment. I think that there are a few possible situations that could occur:
* Subsetting with lapse rate cube. This currently fails with a "Cubes do not share or originate from the same grid, so cannot be used together" error, as the subset forecast doesn't match the grid hash previously generated on the neighbour cube.
* No subsetting with lapse rate cube. If the cubes are set up, so that the spot forecast and neighbour cube have a grid hash matching with the gridded lapse rate cube, then the code seems to run through in principle. Does this mean that a spot lapse rate file would be supported in principle?
* Subsetting with fixed lapse rate. Similar to when using a lapse rate cube, this currently fails as the subset forecast doesn't match the grid hash.
* No subsetting with fixed lapse rate. The code seems to run successfully in this case.

Therefore, I think that main thing is just to make it clear, in docstrings and in the code, whether the application of lapse rate correction (with a (gridded or spot) lapse rate cube or a fixed lapse rate) to site forecasts (with or without subsetting) is supported, as I don't think this is explicitly handled right now.